### ISSUE TITLE
How to use different d3d11 color spaces on Windows 10 for different content?

### ISSUE THREAD

**Issue by Doofussy2** [mpv-player/mpv#7346]
With the improved HDR/CSP settings, how can I set one colorspace for srgb and one for pq? I run Windows 10 in HDR10, so I can have correct HDR playback. mpv automatically detects and sets the colorspace to pq. This greatly darkens SDR content. I've been using auto-profiles to brighten SDR content, but it still isn't right. VLC has no problem with this. It plays SDR content, bright and sharp. If I use `--d3d11-output-csp=srgb`, it is then played back correctly, but HDR is messed up. Setting different colorspaces using auto-profiles, doesn't work. Whatever I set at the top is applied to everything.

Forgive the crappy photo, but here's what I'm referring to:
![IMG_20200110_215344_MP_1 1](https://user-images.githubusercontent.com/15950919/72199776-55572500-33f5-11ea-9aa1-016993c11e66.jpg)

Is there a way to play SDR content correctly when in pq?

**Comment by haasn** [comment]
I'm not sure I understand the question. What do you mean by "use different color spaces"?

**Comment by Doofussy2** [comment]
Switch between d3d11-output-csp=srgb and d3d11-output-csp=pq based on the content I'm playing.

**Comment by Doofussy2** [comment]
Could bit depth be the cause of the picture being darker? I'll have to test that.

**Comment by Doofussy2** [comment]
Changing the bit depth made no difference. But using srgb csp, does.

**Comment by Doofussy2** [comment]
Any suggestions? Or is it not possible?

**Comment by Doofussy2** [comment]
Ok, looks like I figured it out. I thought I had already tried this, but ....

Just had to use this, generally `--d3d11-output-csp=no`, and the create an auto-profile for bt.2020, and add this line `--d3d11-output-csp=pq`

Not elegant, but appears to be functional.

**Comment by Doofussy2** [comment]
I do think a more innate solution would be better.

**Comment by Doofussy2** [comment]
BAH! That doesn't work, either!

**Comment by Doofussy2** [comment]
So I guess that means you can't even disable using the swapchain csp?

**Comment by Doofussy2** [comment]
This is actually an issue now, and not a question.

**Comment by garoto** [comment]
Post your auto-profiles.lua condition.

**Comment by Doofussy2** [comment]
```
## VIDEO ##
gpu-api=auto
hwdec=auto
d3d11-output-csp=srgb
hwdec-codecs=all
opengl-pbo
scale=ewa_lanczossharp

[HDR]
profile-desc=cond:p["video-params/primaries"]=="bt.2020"
d3d11-output-csp=pq
```

With this, srgb is set for everything, and doesn't change.  It would appear that the d3d csp can only be set, once.

**Comment by Doofussy2** [comment]
Ideally, I would like to be able disable d3d csp, only enabling it for HDR stuff. That way I can still use an icc for my SDR stuff. As is, the icc clashes with the d3d csp. But at the least, I want to not use pq for SDR stuff.

**Comment by Doofussy2** [comment]
Can anyone get this to work?

**Comment by Doofussy2** [comment]
Ok, so it looks like cranking the brightness up 8 points, is my only option. So I'll close this.

**Comment by Doofussy2** [comment]
Now I've got it. Adjust the gamma. I hadn't thought of that. Raising the gamma to 15, takes care of business.

**Comment by Doofussy2** [comment]
But this is probably the better way.

```
gamma-factor=1.2

[HDR]
profile-desc=cond:p["video-params/primaries"]=="bt.2020"
gamma-factor=1.0

```

**Comment by wm4** [comment]
> With this, srgb is set for everything, and doesn't change. It would appear that the d3d csp can only be set, once.

Runtime changes are probably not implemented.

Also it looks like you're using autoprofiles.lua, which I originally wrote, but deleted ages ago. I deleted it for a reason.

**Comment by Doofussy2** [comment]
Would runtime changes allow setting d3d csp in auto profiles?

There are a few auto profile scripts, which one is recommended to use? Argon's?

https://github.com/Argon-/mpv-config/blob/master/scripts/auto-profiles-functions.lua

**Comment by Doofussy2** [comment]
I think I've found my happy place, for this. And I figured out some great upscaling, too. These pics are 720p scaled to 2160. (Upscaling stuff [here](https://artoriuz.github.io/mpv_upscaling.html), for those interested). But the color quality is really pretty good.

```
saturation=5
contrast=8
gamma=14
```

![Pic 01](https://user-images.githubusercontent.com/15950919/74128497-2cf84d00-4b92-11ea-9295-ecc5cebc44b9.jpg)

![Pic 02](https://user-images.githubusercontent.com/15950919/74128542-3ed9f000-4b92-11ea-8763-fe578ab32da5.jpg)

![Pic 03](https://user-images.githubusercontent.com/15950919/74128563-48635800-4b92-11ea-8d69-a30672be5070.jpg)

![Pic 04](https://user-images.githubusercontent.com/15950919/74128570-4e593900-4b92-11ea-9ee6-c364aea72087.jpg)

![Pic 05](https://user-images.githubusercontent.com/15950919/74128582-54e7b080-4b92-11ea-9d5d-d2b0be8316f5.jpg)

**Comment by Doofussy2** [comment]
I just wanted to quickly update this, for reference. Thanks to @haasn's great work with this commit  https://github.com/mpv-player/mpv/commit/ef6bc8504a945eb6492b8ed46fd5a1afaaf32182, the picture quality has improved. It's still a little dark, but now a simple gamma bump to `--gamma=10` is all that's needed. The white point is now good, and color is how it should be.

**Comment by Doofussy2** [comment]
I'm re-opening this. I just realized that the gamma is low on everything. I was watching an HDR movie using an Nvidia Shield TV (connected to the same display, with the same calibration). When I switched over to using mpv on the PC, it was notably darker. Some of the colors became indiscernible. I raised the gamma by 10 points, and then it matched more closely what I was seeing on the Shield. The difference is significant. 

Ah, but when playing in it in VLC I get the same darkness. So possibly a Windows thing or Nvidia? Either way, raising the gamma to 10 or 12, works quite nicely.
